support/upsert/loopdetector.go

```diff
@@ -31,7 +31,7 @@ const LoopDetectorWarningMessage = "WARNING: Object got updated more than one ti
 // in the future.
 // Once we did a no-op update, we will ignore the object because we assume that if we have
 // a bug in the defaulting, we will end up always updating.
-const updateLoopThreshold = 1
+const updateLoopThreshold = 2
```

Comment:
@enxebre the issue is that for just updates, the loop checker would always fail. It would set the updateEventCount to 1 here (https://github.com/openshift/hypershift/blob/b98bc43467a73921891ede7d6477757b647f1985/support/upsert/loopdetector.go#L64) and then check whether it was less than 1. It's not an issue for other resources because we first create them. However in this PR, I am reconciling the default service account which we don't create, just update.